week3/d01/Assignment_3_1_1/ex4.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include <exception>
+#include <stdexcept>
+#include <string>
+
+
+/*
+**  Takes a string and loops through each character, throws an exception if
+**  any chacater is not an alphabet
+*/
+void process_data(std::string& str)
+{
+    for (auto& c : str)
+    {
+        if (!isalpha(c))
```

Comment:
[Like all other functions from cctype, the behavior of std::isalpha is undefined if the argument's value is neither representable as unsigned char nor equal to EOF. To use these functions safely with plain chars (or signed chars), the argument should first be converted to unsigned char:](https://en.cppreference.com/w/cpp/string/byte/isalpha)